File: data/src/test/java/com/tonitealive/app/data/DefaultMessageProviderTest.java
package com.tonitealive.app.data;

import com.tonitealive.app.TestConstants;
import com.tonitealive.app.domain.MessageId;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;
import org.robolectric.annotation.Config;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(RobolectricTestRunner.class)
@Config(constants = BuildConfig.class, sdk = {TestConstants.SDK_VERSION})
public class DefaultMessageProviderTest {

    private DefaultMessageProvider messageProvider;

    public DefaultMessageProviderTest(){}

    @Before
    public void setup() {
        messageProvider = new DefaultMessageProvider(RuntimeEnvironment.application);
    }

    @After
    public void teardown() {
        Robolectric.reset();
    }

    @Test
    public void getMessage_shouldReturnMessage() {
        // With
        MessageId messageId = MessageId.SIGN_IN_SUCCESS;
        int messageCode = R.string.signin_succeeded;
        String targetMessage = RuntimeEnvironment.application.getString(messageCode);

        // When
        String message = messageProvider.getMessage(messageId);

        //  Then
        assertThat(message).isEqualTo(targetMessage);
    }

}

File: data/src/test/kotlin/com/tonitealive/app/data/DefaultMessageProviderTest.kt
package com.tonitealive.app.data

import com.tonitealive.app.SDK_VERSION
import com.tonitealive.app.domain.MessageId
import org.assertj.core.api.Assertions.assertThat
import org.junit.After
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.Robolectric
import org.robolectric.RobolectricTestRunner
import org.robolectric.RuntimeEnvironment
import org.robolectric.annotation.Config

@RunWith(RobolectricTestRunner::class)
@Config(constants = BuildConfig::class, sdk = intArrayOf(SDK_VERSION))
class DefaultMessageProviderTest {

    lateinit var messageProvider: DefaultMessageProvider

    @Before
    fun setup() {
        messageProvider = DefaultMessageProvider(RuntimeEnvironment.application)
    }

    @After
    fun teardown() {
        Robolectric.reset()
    }

    @Test
    fun getMessage_shouldReturnMessage() {
        // With
        val messageId = MessageId.SIGN_IN_SUCCESS
        val messageCode = R.string.signin_succeeded
        val targetMessage = RuntimeEnvironment.application.getString(messageCode)

        // When
        val message = messageProvider.getMessage(messageId)

        //  Then
        assertThat(message).isEqualTo(targetMessage)
    }

}